tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc

```diff
@@ -5445,97 +5578,84 @@ Status ConvertFullyConnectedHelper(OpConverterParams* params,
+          "MatMul with 2D tensors requires explicit batch mode, or op(A)=None"
+          " and B as weight.");
+    }
   }
 
-  // If B is a tensor, then it must be at least 3D in TF,
-  // or TRT won't be able to handle the multiply correctly.
-  if (input_b.is_tensor() && input_b.GetTrtDims().nbDims < 2) {
-    return errors::InvalidArgument(
-        "Second input must either be a constant, or contain at least 2 "
-        "non-batch dimensions.");
-  }
   if (params->validation_only) return Status::OK();
 
-  // If an FC layer can be used and would be faster, use that instead.
-  const bool can_use_fc =
-      !transpose_a && input_a.is_tensor() && input_b.is_weights();
-  const bool should_use_fc = can_use_fc && input_a.GetTrtDims().nbDims >= 3 &&
-                             input_b.GetTrtDims().nbDims == 2;
-  // If int8 is specified, FC must be used unless it is not compatible, as MM
-  // does not support int8 at this time.
-  if (should_use_fc || (can_use_fc && params->converter->precision_mode() ==
-                                          TrtPrecisionMode::INT8)) {
-    return ConvertFullyConnectedHelper(
-        params, input_a.tensor(), input_b.weights(), transpose_b, node_def);
-  }
-
-  const auto get_matrix_op = [](nvinfer1::ITensor* in,
-                                bool transpose) -> nvinfer1::MatrixOperation {
-    return (in->getDimensions().nbDims < 2) ? nvinfer1::MatrixOperation::kVECTOR
-           : (transpose) ? nvinfer1::MatrixOperation::kTRANSPOSE
-                         : nvinfer1::MatrixOperation::kNONE;
-  };
+  bool converted = false;
+  TF_RETURN_IF_ERROR(ConvertFullyConnectedHelper(params, input_a, input_b,
+                                                 transpose_a, transpose_b,
+                                                 node_def, &converted));
+  if (converted == true) return Status::OK();
 
-  // If the MatMul operand is a constant, applies transposes at conversion-time
-  // as necessary. If the operand is a tensor, does nothing. If required
-  // transposes were applied, sets transpose to false.
+  // Add constant layer for weight operand, otherwise do nothing.
   const auto prepare_matmul_operand =
       [&params](TRT_TensorOrWeights operand,
                 bool* transpose) -> nvinfer1::ITensor* {
     if (operand.is_tensor()) {
       return operand.tensor();
     } else {
-      TRT_ShapedWeights weights(operand.weights().TrtDType());
-      if (*transpose) {
-        weights = params->weight_store->GetTempWeights(operand.weights());
-        ReorderCKtoKC(operand.weights(), &weights);
-        // Weights have been transposed, can set transpose to false
-        *transpose = false;
-      } else {
-        weights = operand.weights();
-      }
-      return params->converter->CreateConstantLayer(weights, weights.shape_);
+      return params->converter->CreateConstantLayer(operand.weights(),
+                                                    operand.GetTrtDims());
```

Comment:
The "tranpose" parameter is no longer use. Here is what I would suggest:
. Rename the lambda  convert_to_itensor
. We probably don't need the comment as it only repeat the code.